docker-compose.yml

```diff
@@ -0,0 +1,65 @@
+#
+# Basic version of stack, for Github
+#
+version: "3.3"
+services:
+  sequencescape_server:
+    build: .
+    image: sequencescape_local_image
+    depends_on:
+      - mysql_server
+    environment:
+      - DBHOST=mysql_server
+      - DBPORT=3306
+      - RESET_DATABASE=$RESET_DATABASE
+    entrypoint: "bash -c ./entrypoint.sh"
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:3000/health"]
+      interval: 30s
+      timeout: 10s
+      retries: 4
+
+  sequencescape_worker:
+    image: sequencescape_local_image
+    depends_on:
+      - sequencescape_server
+    environment:
+      - DBHOST=mysql_server
+      - DBPORT=3306
+    entrypoint: "bash -c 'bundle exec rake jobs:work'"
+    healthcheck:
+      test: "ps -ef | grep -q jobs"
+      interval: 10s
+      retries: 5
+
+  mysql_server:
+    image: mysql:5.7
+    environment:
+      - MYSQL_ALLOW_EMPTY_PASSWORD="yes"
+    healthcheck:
+      test: "/etc/init.d/mysql status"
+      interval: 30s
+      retries: 120
```

Comment:
This MySQL server might conflict with the one you might be running for Lighthouse and Crawler.  We might have to think about this more carefully, or at least not start this server on a local dev machine unless a flag is set, for example.

Reply:
This would be like an option that can be used along with native;  I was thinking on leaving the decision of what would be needed to later to see what it would be the best hybrid approach for us; about doing this this way I thought this had the advantage that it will work independently to what anybody has in local; you should be able to run a full complete Sequencescape even if your local version of mysql is broken (of course as long as Docker works), so to have a full safe option to work around the problems with native installations. But this is an initial config, I think it can be modified in other stories/changes to adapt to what is better for us.

Reply:
Yes, I dont have a global solution for it. Is the same problem if you want to connect the Rabbitmq from Sequencescape with the unified_warehouse consumers, for example.  We don't have a common configuration to have all projects to run together, but this is very early stage.
The only temporal option I can think of is to have different profiles of docker-compose and so you can use the one is better for your situation, but keeping a default one that should work with the easier setup. This solution for the moment has 2 profiles: 
1) docker-compose up : runs docker-compose.yml + docker-compose.override.yml so it will start everything in local
2) docker-compose -f docker-compose.yml up : runs only docker-compose.yml so it will work in github

So it could be added another like docker-compose-local.yml that could be run like: ```docker-compose -f docker-compose-localdev.yml up```, would it work for you?

Reply:
According to Docker docs: https://docs.docker.com/compose/extends/

They say that you could have a production file `docker-compose.prod.yml` which you could use in the build config for the GitHub workflow.  Ideally that would have the MySQL service in it while the un-prefixed one does not have MySQL.  Then the developer can choose to add the production file to their `docker-compose up` if they want to have the MySQL, but most of the time it is expected you have a MySQL instance already available.

Reply:
I agree but I would do it the other way. I would keep the setup as easier as possible to have all running in one command, so anybody can run and test Sequencescape in the easier way possible (if anybody wants to) and then have a separate local configuration for custom development for working with any custom flags there if they are needed.

Reply:
docker-compose up -> with mysql
docker-compose -f <localtemplate> up -> without mysql

Reply:
If you can do that, that also seems fine.  The problem I was having with docker-compose create your MySQL was that it kept destroying the container and the data associated with it, so you can unexpectedly lose all your database contents that you've been setting up to test an issue because of one quick slip.  That's why I made it a completely separate docker-compose for just the database dependencies in Lighthouse and Crawler.

Reply:
In these files ```docker-compose.override.yml``` includes a volume for mysql that refers to local folder so you can destroy the container and create it again but the data remains there.
Ok, I'll add the localdev template.

Reply:
Having discussed offline, there has been some misunderstanding on my part.  The port for MySQL is not being exposed and therefore won't cause an issue when trying to run alongside a local installation or another Docker container for MySQL.  I would revert the change made for the additional docker-compose file.  If it turns out that it's hard to access MySQL as a developer when the port isn't exposed by default, we should reconsider, but for now I think this is fine.